src/Cli/dotnet/commands/dotnet-test/TestingPlatformCommand.cs

```diff
@@ -67,28 +67,42 @@ public int Run(ParseResult parseResult)
                 });
             }
 
+            _args = [.. parseResult.UnmatchedTokens];
             VSTestTrace.SafeWriteTrace(() => $"Wait for connection(s) on pipe = {_pipeNameDescription.Name}");
             _namedPipeConnectionLoop = Task.Run(async () => await WaitConnectionAsync(_cancellationToken.Token));
 
-            bool containsNoBuild = parseResult.UnmatchedTokens.Any(token => token == CliConstants.NoBuildOptionKey);
-            List<string> msbuildCommandlineArgs = [$"-t:{(containsNoBuild ? string.Empty : "Build;")}_GetTestsProject",
-                 $"-p:GetTestsProjectPipeName={_pipeNameDescription.Name}",
-                    "-verbosity:q"];
-
-            AddAdditionalMSBuildParameters(parseResult, msbuildCommandlineArgs);
-
-            if (VSTestTrace.TraceEnabled)
+            if (parseResult.HasOption(TestCommandParser.TestModules))
             {
-                VSTestTrace.SafeWriteTrace(() => $"MSBuild command line arguments: {string.Join(" ", msbuildCommandlineArgs)}");
-            }
+                // If the module path pattern(s) was provided, we will use that to filter the test modules
+                string testModules = parseResult.GetValue(TestCommandParser.TestModules);
 
-            ForwardingAppImplementation msBuildForwardingApp = new(GetMSBuildExePath(), msbuildCommandlineArgs);
-            int testsProjectResult = msBuildForwardingApp.Execute();
+                // If the root directory was provided, we will use that to search for the test modules
+                // Otherwise, we will use the current directory
+                string rootDirectory = parseResult.GetValue(TestCommandParser.TestModulesRootDirectory) ?? Directory.GetCurrentDirectory();
```

Comment:
We should check if the `TestModulesRootDirectory` exists or return a message and faile.